### PR TITLE
Bugfix/remove dollar before method or header

### PR DESCRIPTION
**Pull request by Dhroov7**
Solving Issue- #1 
Removing unnecessary dollars in curl command and also added test.
@abhijitkane Please review.

### CODE REVIEW

File: test/conversion.test.js [hunk 1]
```diff
@@ -27,6 +27,21 @@ describe('Curl converter should', function() {
 		});
 	});
 
+	it('[Github: #1]: remove all the $ which are present before a string method or headers', function (done) {
```
Comment (shreys7) [inline test/conversion.test.js]: We should have two different tests:
1. one for the exact bug fix stating that the convert function doesn't throw error when there is a $ before method
2. As you have also fixed that the values of headers and body doesn't contain a $ after importing, separate test for that too

File: src/lib.js [hunk 2]
```diff
@@ -193,31 +193,43 @@ var curlConverter = {
         return arr.join("&");
     },
 
+    sanitizeCurlArgs: function(curlString) {
```
Comment (shreys7) [inline src/lib.js]: this should be renamed with just sanitizeArgs or similar name. This is for sanitizing any bash string and not the specific curlString as discussed earlier. Hence the input argument name should also be changed to bash string ?

File: test/unit.test.js [hunk 3]
```diff
@@ -19,11 +19,27 @@ var Converter = require('../src/lib.js'),
...
 
 			var result = validate('invalid curl');
 			expect(result.result).to.be(false);
+
+			done();
 		});
 	});
 
+	describe('sanitizeArgs function should', function () {
+		it('remove all unnecessary $ from curl string', function (done) {
```
Comment (shreys7) [inline test/unit.test.js]: Similar comment, the test name should be from bash string and not curl string.

File: test/unit.test.js [hunk 4]
```diff
@@ -19,11 +19,27 @@ var Converter = require('../src/lib.js'),
...
+			let curlString = `curl -i -s -k  -X $'POST' \
+			-H $'Host: example.com.br' -H $'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:68.0) Gecko/20100101 Firefox/68.0'\
+			$'https://example.com.br/login.html'`
```
Comment (shreys7) [inline test/unit.test.js]: Shouldn't we check for this $ being getting removed too as we are checking for all the $ signs ?
There doesn't seem to be any test for this.

File: src/lib.js [hunk 5]
```diff
@@ -193,31 +193,43 @@ var curlConverter = {
...
+        var sanitizedArgs = _.map(_.filter(argv, function(arg) { return !_.isEmpty(arg) }), function (arg) {
+          if (_.isObject(arg) && arg.op === 'glob') {
+            return arg.pattern
+          }
+          else if (arg.op && arg.op.startsWith('$') && arg.op.length > 3) {
+            // in the app, certain headers like -H $'cookie: abc' are treated as operators
+            // converting the arg to cookie: abc instead of op: $'cookie: abc'
+            return arg.op.substring(2, arg.op.length-1);
+          }
+          else if (arg.startsWith('$') && arg.length > 2) {
```
Comment (abhijitkane) [inline src/lib.js]: will `-X` be part of `arg`?

File: test/conversion.test.js [hunk 6]
```diff
@@ -27,6 +27,18 @@ describe('Curl converter should', function() {
 		});
 	});
 
+	it('[Github: #1]: not throw an error for having $ before method', function (done) {
+		convert({
+			type: 'string',
+			data: `curl -i -s -k  -X $'POST' \
```
Comment (abhijitkane) [inline test/conversion.test.js]: if you're testing for `$ before method`, use the *simplest* input that tests for that. No headers/body needed.

File: src/lib.js (hunk 5 above)
Comment (abhijitkane) [reply src/lib.js]: @Dhroov7 ^
This is for the `converting the arg to -X 'POST'` comment.

Comment (abhijitkane) [reply src/lib.js]: also add a link to the spec/RFC which talks about what $'POST' is supposed to mean

Comment (Dhroov7) [reply src/lib.js]: @abhijitkane Okay!!

Comment (abhijitkane) [reply src/lib.js]: @Dhroov7 There are 2 comments before the RFC one too.

File: src/lib.js [hunk 7]
```diff
@@ -193,31 +193,44 @@ var curlConverter = {
...
+            // removing $ before every string like -X $'POST' and
+            // converting the arg to -X 'POST'
+            // link of RFC- https://tiswww.case.edu/php/chet/bash/bashref.html#ANSI_002dC-Quoting
```
Comment (abhijitkane) [inline src/lib.js]: This is technically not an RFC. See if you can find a more authoritative doc. Check https://unix.stackexchange.com/questions/48106/what-does-it-mean-to-have-a-dollarsign-prefixed-string-in-a-script. This link should be included in the comments too.

File: test/conversion.test.js [hunk 8]
```diff
@@ -27,6 +27,16 @@ describe('Curl converter should', function() {
 		});
 	});
 
+	it('[Github: #1]: not throw an error for having $ before method', function (done) {
```
Comment (abhijitkane) [inline test/conversion.test.js]: This isn't only for the method. Either we test only for method, or we mention method and URL in the test name.

File: src/lib.js (hunk 7 above)
Comment (Dhroov7) [reply src/lib.js]: But both links have the same doc.

File: test/unit.test.js (hunk 4 above)
Comment (shreys7) [reply test/unit.test.js]: We don't need to provide the arguments curl, -i, -s etc. As mentioned by @abhijitkane , we should be using the simplest input for testing the exact thing you are checking for.

File: src/lib.js (hunk 5 above)
Comment (Dhroov7) [reply src/lib.js]: @abhijitkane Yes, `-X` will be part of `arg`.

File: test/conversion.test.js [hunk 9]
```diff
@@ -27,6 +27,16 @@ describe('Curl converter should', function() {
...
+		convert({
+			type: 'string',
+			data: `curl -X $'POST' 'https://example.com.br/login.html'`
+		}, function (err, result) {
+			expect(result.result).to.equal(true);
```
Comment (abhijitkane) [inline test/conversion.test.js]: Let's also check for the actual request.method being POST. The other test that's added is a unit test. We should test the e2e flow here - which is not limited to `result` being true.

**Pull request merged**